src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,141 @@
-import { Component } from 'react';
-
-type Props = {};
-type State = {};
-
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
-
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
-    );
-  }
-}
+import React, { useState } from 'react';
+import './NewMovie.scss';
+import className from 'classnames';
+
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setimdbUrl] = useState('');
+  const [imdbId, setimdbId] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrlDirty, setImgUrlDirty] = useState(false);
+  const [errorImgUrl, setErrorImgUrl] = useState('Required field');
+  const [imdbUrlDirty, setImdbUrlDirty] = useState(false);
+  const [errorImdbUrl, setErrorImdbUrl] = useState('Required field');
+
+  const clearForm = () => {
+    setTitle('');
+    setImgUrl('');
+    setimdbId('');
+    setimdbUrl('');
+    setDescription('');
+    setImgUrlDirty(false);
+    setImdbUrlDirty(false);
+  };
+
+  const newMovie = {
+    title,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+    description,
+  };
+
+  const handleSubmit = () => {
+    onAdd(newMovie);
+    clearForm();
+  };
+
+  const urlHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    switch (event.target.name) {
+      case 'imgUrl': setImgUrlDirty(true);
+        break;
+      case 'imdbUrl': setImdbUrlDirty(true);
+        break;
+      default: setImgUrlDirty(false);
+        setImdbUrlDirty(false);
+        break;
+    }
+  };
+
+  const placeHandlerValidate = (
+    event: React.ChangeEvent<HTMLInputElement>, state: (str: string) => void,
```

Comment:
```suggestion
    event: React.ChangeEvent<HTMLInputElement>, setErrorFunction: Dispatch<SetStateAction<string>> => void,
```
Для этой функции есть типы в react